File: CLASE IV - Listas/mejorado/listas_int/c/nodo.c
#include <stdio.h>
#include <stdlib.h>
#include "nodo.h"



NodoPtr crearNodo(int dato, NodoPtr siguiente){


    NodoPtr nodo = (NodoPtr) malloc(sizeof(struct Nodo));

    nodo->dato = dato;
    nodo->siguiente = siguiente;

    return nodo;

};

void mostrarNodo(NodoPtr nodo){

    printf("{ %d }", nodo->dato);

};


void liberarNodo(NodoPtr nodo){

    free(nodo);


};

File: CLASE IV - Listas/listas_int/golang/nodo.go
package main

import "fmt"

type Nodo struct {
	dato      int
	siguiente *Nodo
}

func crearNodo(dato int, siguiente *Nodo) *Nodo {
	return &Nodo{dato: dato, siguiente: siguiente}
}

func mostrarNodo(nodo *Nodo) {
	fmt.Printf("{ %d }", nodo.dato)
}

